flink-core/src/main/java/org/apache/flink/configuration/YamlParserUtils.java

```diff
@@ -0,0 +1,227 @@
+     * data.
+     *
+     * <p>NOTE: It will skip escaping string values in order to avoid ambiguity.
+     *
+     * @param value The value to be converted.
+     * @return The string representation of the value in YAML syntax.
+     */
+    public static String toConfigurationDataString(Object value) {
+        if (value instanceof String) {
+            return value.toString();
+        } else {
+            return toYAMLString(value);
+        }
+    }
+
+    /**
+     * Converts the given value to a string representation in the YAML syntax. This method uses a
+     * YAML parser to convert the object to YAML format.
+     *
+     * <p>The resulting YAML string may have line breaks at the end of each line. This method
+     * removes the line break at the end of the string if it exists.
+     *
+     * <p>Note: This method may perform escaping on certain characters in the value to ensure proper
+     * YAML syntax.
+     *
+     * @param value The value to be converted.
+     * @return The string representation of the value in YAML syntax.
+     */
+    public static synchronized String toYAMLString(Object value) {
+        try {
+            String output = yaml.dump(value);
+            // remove the line break
+            String linebreak = dumperOptions.getLineBreak().getString();
+            if (output.endsWith(linebreak)) {
+                output = output.substring(0, output.length() - linebreak.length());
+            }
+            return output;
+        } catch (MarkedYAMLException exception) {
+            throw wrapExceptionToHiddenSensitiveData(exception);
+        }
+    }
+
+    public static synchronized <T> T convertToObject(String value, Class<T> type) {
+        try {
+            return yaml.loadAs(value, type);
+        } catch (MarkedYAMLException exception) {
+            throw wrapExceptionToHiddenSensitiveData(exception);
+        }
+    }
+
+    /**
+     * This method is simply mocked the MarkedYAMLException#getMessage() to hidden sensitive data.
+     */
+    private static YAMLException wrapExceptionToHiddenSensitiveData(MarkedYAMLException exception) {
+        StringBuilder lines = new StringBuilder();
+        String context = exception.getContext();
+        Mark contextMark = exception.getContextMark();
+        String problem = exception.getProblem();
+        Mark problemMark = exception.getProblemMark();
+
+        if (context != null) {
+            lines.append(context);
+            lines.append("\n");
+        }
+        if (contextMark != null
+                && (problem == null
+                        || problemMark == null
+                        || contextMark.getName().equals(problemMark.getName())
+                        || (contextMark.getLine() != problemMark.getLine())
+                        || (contextMark.getColumn() != problemMark.getColumn()))) {
+            lines.append(hiddenSensitiveDataInMark(contextMark));
```

Comment:
Could you share an example to show what the exception looks like with/without this wrapping?

Reply:
Sure, I've updated the javaDoc to include some sample code for clarification.